### PR TITLE
[NDPE-69] Script assets and page flow examples

### PR DESCRIPTION
**Pull request by ghost**
As the example prototypes demonstrating page flow required the use of the frontend library's radio button pattern, Javascript support for that pattern had to be added first.

The first commit moves the source code view for the UI library inline, as discussed.

The next commit introduces custom script support to the kit modelled after the custom SCSS support. `assets/scripts/app/app.js` is provided for user scripting whereas `assets/scripts/nhsuk/nhsuk.js` is for kit scripting and contains the minimal code necessary to manage the radio button pattern.

Gulp tasks copy the scripts into the `/public` directory during build. I added a short delay to the BrowserSync config to smooth out the reloading process.

Finally, two example prototypes are added to demonstrate page flow using radio buttons, with "view source" capability added to all the example prototypes.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -133,7 +133,8 @@ utils.findAvailablePort(app, function (port) {
         ghostmode: false,
         open: false,
         notify: false,
-        logLevel: 'error'
+        logLevel: 'error',
+        reloadDelay: 1000
```
Comment (visualunderground) [inline server.js:137]: Is the delay needed due to script compilation?

Comment (ghost) [reply server.js:137]: Not really - it's more that the styles take a while to compile and without this delay I was getting poor page loads

Comment (ghost) [reply server.js:137]: Which may be due to the change I made to the gulp watch tasks - I'll check

Comment (ghost) [reply server.js:137]: If I switch back to using runSequence in the gulp.watch task and take out the reload delay, I still get bad page reloads. Looks like adding in the scripts just tipped the timing balance on reloads, so the delay is necessary now.

File: app/views/index.hbs [hunk 2]
```diff
@@ -5,4 +5,5 @@
 {{#if showDocumentation}}
 <p><a href="/docs/guides/index.md">Documentation</a></p>
 <p><a href="/components">UI Library</a></p>
+<p><a href="/examples/index">Examples</a></p>
```
Comment (visualunderground) [inline app/views/index.hbs]: Examples -> Example Prototypes

File: app/views/examples/index.hbs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<h1>NHSUK Prototyping Kit</h1>
+<h2>Example Prototypes</h2>
+<ul>
+  <li><a href="/examples/test">Smoke Test</a>
```
Comment (visualunderground) [inline app/views/examples/index.hbs]: What's a "Smoke test", is there a better name?

File: app/views/index.hbs (hunk 2 above)
Comment (ghost) [reply app/views/index.hbs]: Agreed

File: app/views/examples/index.hbs (hunk 3 above)
Comment (ghost) [reply app/views/examples/index.hbs]: Well, it's an example intended to exercise the basic operation of the kit. I did just call it "Test" before. Perhaps "Kit Test Prototype"?

File: server.js (hunk 1 above)
Comment (visualunderground) [reply server.js:137]: Fair enough

File: app/views/examples/index.hbs (hunk 3 above)
Comment (visualunderground) [reply app/views/examples/index.hbs]: Is it a "Simple example", "Basic use", or "Passing data to templates"?

Comment (ghost) [reply app/views/examples/index.hbs]: Passing Data to Templates - genius!

Comment (ghost) [reply app/views/examples/index.hbs]: I've gone with "Test passing data to a template"

**Review by visualunderground — APPROVED** [review]

**Pull request merged**